### PR TITLE
Feature/007 route audio

### PR DESCRIPTION
**Pull request by raphrel**

### CODE REVIEW

File: Sources/State/PlayingState.swift [hunk 1]
```diff
@@ -112,4 +117,13 @@ public final class PlayingState: PlayerState {
             else { self.playbackStalled(); return }
         context.changeState(state: StoppedState(context: context))
     }
+    
+    private func routeAudioChanged(reason: AVAudioSessionRouteChangeReason) {
+        switch reason {
+        case .oldDeviceUnavailable, .categoryChange, .unknown:
+            context.changeState(state: PausedState(context: context))
+        case .newDeviceAvailable, .wakeFromSleep, .override, .noSuitableRouteForCategory, .routeConfigurationChange:
```
Comment (JiceDessaint) [inline Sources/State/PlayingState.swift]: Maybe put some logs here?
And why not 'default' case?

File: Tests/Services/RouteAudioServiceTests.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    
+                    // ACT
+                    NotificationCenter.default.post(notif)
+                    
+                    // ASSERT
+                    expect(self.routeChangedReason).to(equal(AVAudioSessionRouteChangeReason.oldDeviceUnavailable))
+                }
+            }
+            
+            context("invalid audio route changed reason") {
```
Comment (JiceDessaint) [inline Tests/Services/RouteAudioServiceTests.swift]: To be exhaustive, you have 3 failing tests:
-> notification has no UserInfo
-> value of 'info' dictionary is not a Uint (set a negative number for example)
-> value is a Int not defined for AVAudioSessionRouteChangedReason

You have just test the third case here

File: Sources/State/PlayingState.swift (hunk 1 above)
Comment (raphrel) [reply Sources/State/PlayingState.swift]: I logged in the route audio service. is it better to have information here ?
more readable to understand what case do what thing, no ?

File: Tests/Services/RouteAudioServiceTests.swift (hunk 2 above)
Comment (raphrel) [reply Tests/Services/RouteAudioServiceTests.swift]: I add some tests.
the third is actually false. init with an UInt not define in AVAudioSessionRouteChangedReason still return an "AVAudioSessionRouteChangedReason"

File: Sources/State/PlayingState.swift (hunk 1 above)
Comment (JiceDessaint) [reply Sources/State/PlayingState.swift]: Ok for logging on service. Didn't notice that.
For default, np, it was just a question. I'm OK with that

**Pull request merged**